src/simplify_algebra.cpp

```diff
@@ -851,6 +851,84 @@ struct find_div_const
     }
 };
 
+struct find_zero_div_const
+{
+    auto matcher() const { return match::name("div")(match::arg(1)(match::has_value(0.0f))); }
+
+    [[noreturn]] void apply([[maybe_unused]] module& m,
+                            [[maybe_unused]] const match::matcher_result& r) const
+    {
+        MIGRAPHX_THROW("ERROR: Matched division by zero in pass");
```

Comment:
I dont think we should throw an exception here. This could be code that would be eliminated by DCE so it doesn't make sense to warn. Also, simplify_algebra is for doing transformation.

I think we need to put such validations in a seperate pass or function somewhere, but we may not want to just throw an error. If this div by zero came from the user model then we would point to where the error came from in the IR.

Reply:
I see, thats where my apprehension was with this case since this should be flagged as an error if say somehow our math/simplifications lead to a division by 0. 

Should I look at part of DCE to flag this instead then and remove this for now?

Reply:
We dont really have a place for this kind of checking but it probably needs to be in a seperate function we call. I think it would be best to put this in another PR.